### ISSUE TITLE
[bug]: Simple taproot channel inactive after 6 confs

### ISSUE THREAD

**Issue by hieblmi** [lightningnetwork/lnd#8065]
### Background

Simple taproot channels stay inactive if the initiator disconnects after the channel opening and reconnects after the channel has reached 6 confirmations. The following error appears in the logs upon reconnection:
```
2023-10-05 11:42:53.684 [WRN] PEER: Peer(03a01da97af71f7859cd1b2b6a70f221ecd49f6eabf0bf1e267bae9570b47232a6): Unable to find our forwarding policy for channel redacted:0, using default values
2023-10-05 11:47:16.999 [WRN] PEER: Peer(03a01da97af71f7859cd1b2b6a70f221ecd49f6eabf0bf1e267bae9570b47232a6): Unable to find our forwarding policy for channel redacted:0, using default values
2023-10-05 15:54:01.694 [WRN] CHBU: Replacing disk backup for ChannelPoint(redacted:0) w/ newer version
2023-10-05 15:54:01.906 [WRN] HSWC: ChannelLink(redacted:0): error when syncing channel states: unable to init remote nonce: pending verification nonce is not set 
2023-10-05 15:54:01.906 [ERR] HSWC: ChannelLink(redacted:0): failing link: unable to synchronize channel states: unable to init remote nonce: pending verification nonce is not set with error: unable to resume channel, recovery required
```

After a `lnd` restart however, the channel turns active.

### Your environment

`v0.17.0-beta`

### Steps to reproduce

This `itest` reproduces the issue: https://github.com/lightningnetwork/lnd/pull/8070

In short:

1. Alice opens a simple taproot chan to Bob
2. Then she disconnects from Bob
3. The channel is confirmed on-chain(6 confs)
4. Alice reconnects but channel stays inactive. 
5. Only after alice restarts does the channel turn active

### Expected behaviour

After Alice reconnects the channel negotiation should continue to activate the channel.

### Actual behaviour

Instead the channel stays inactive.
